HIP-0006.md

```diff
@@ -0,0 +1,170 @@
+the name owner from canceling an in-progress name ownership transfer. Since there is a
+small amount of interactivity required in the protocol, the name owner must also publish
+"where" the auction will take place. The auction "location" is just some internet endpoint
+where bidders can submit bids. It can be a private email address (in which case the name owner
+may choose to keep all bids confidential or truly blind), or it can be a public server
+that announces every new bid as they arrive.
+
+### Auction BID
+
+A user that wants to bid on the name constructs a partially-signed transaction in
+the following format:
+
+```
+vin:
+  0: current name owner, encumbered by Swap Script
+  1: A UTXO controlled by the bidder of some value V
+vout:
+  0: TRANSFER to the bidder's address
+```
+
+Input 1 is signed with `SIGHASH_ANYONECANPAY | SIGHASH_SINGLEREVERSE`. This commits
+the incoming payment to the name transfer. Note that Input 0 is not signed yet,
+but it must be the current name-owning UTXO or the linked output 0 would be invalid.
+
+Note that this transaction has no change output. Because the bidder is signing
+with single-reverse, they can't specify a change output in addition to the name
+transfer output. If they were to sign with `SIGHASH_ALL` then the next step of
+the protocol would not be possible. If they signed with `SIGHASH_SINGLE` they could
+commit to a change address, but not the name transfer.
+
+Having no change output means the INPUT coin they are spending MUST BE THE VALUE
+OF THE BID THEY INTEND TO PLACE ON THE NAME. This is a big inconvenience
+that can be solved in at least these two ways:
+
+- The user prepares their wallet in advance and collects UTXOs with a range
+of values (1 HNS, 10 HNS, 100 HNS, 1000 HNS, etc).
+
+- When the user wants to bid, their wallet creates a "perfect" UTXO by spending
+to itself, creating a coin with the exact value desired. This output can be used
+(even before it is confirmed) in this protocol.
+
+### Auction FINALIZE
+
+The name owner collects partially-signed bids and after some time, declares the highest
+value bid the winner. The name owner completes the auction by adding a payment output
+to the partially-signed transaction, signing and broadcasting:
+
+```
+vin:
+  0: current name owner, encumbered by Swap Script
+  1: A UTXO controlled by the bidder of some value V
+vout:
+  0: TRANSFER to the bidder's address
+  1: payment to the name-owner
+```
+
+The name owner pays themselves from the fully funded transaction, and signs input 0
+with `SIGHASH_ALL`. This transaction is a valid atomic swap and can be broadcast to the network.
+Because the locking script prevents all other name actions, the auction winner can
+`FINALIZE` this name transfer any time after the 288-block lockout without any other
+obligation.
+
+Note that since the bidder creates the INPUT and the name owner creates the OUTPUT,
+the name owner MUST also pay the miner fee for this transaction. They do so by reducing
```

Comment:
Is the "owner pays mining fee" caveat important enough to be mentioned again in a separate section?

Reply:
👍